src/pubsub.c

```diff
@@ -95,6 +95,28 @@ pubsubtype pubSubShardType = {
     .messageBulk = &shared.smessagebulk,
 };
 
+uint64_t clientHash(const void *key) {
+    return ((client *)key)->id;
+}
+
+int clientKeyCompare(dict *d, const void *key1, const void *key2) {
+    uint64_t l1,l2;
+    UNUSED(d);
+
+    l1 = ((client *)key1)->id;
+    l2 = ((client *)key2)->id;
+    return l1 == l2;
+}
+
+/* Set dictionary type. Keys are client, values are not used. */
+dictType clientDictType = {
+    clientHash,               /* hash function */
+    NULL,                     /* key dup */
+    NULL,                     /* val dup */
+    clientKeyCompare,         /* key compare */
+    .no_value = 1             /* no values in this dict */
+};
```

Comment:
maybe this should be in server.c? (i imagine we may use it for other client lists in the future)